### PR TITLE
Upgrade to React Navigation 5.0

### PR DESCRIPTION
**Pull request by thumn**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Refactored code for upgrade to React Navigation 5.0
- Separated stack navigators into individual files under `stack_navigators` folder
- Removed `RootStack` and moved `RewardsOverlay` modal to `StoresStack`

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
#52 

### Online sources
[The entire API for React Navigation 5.0](https://reactnavigation.org/docs/upgrading-from-4.x)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
N/A

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Play around with the app and let me know if there are any bugs/noticeable differences from master!
- Make sure the rewards modal still transitions up and down on top of the map screen
- Check that the drawer has the correct username and that the drawer items are accurate
- No screens should be erroring out

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Upgrade to Expo SDK 37

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Played around with basic functionality of the app to make sure the refactoring didn't break anything (but let me know if there's something you catch).

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
N/A

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

**Comment by thumn** [comment]
@12aschen (oops I accidentally tagged you in the wrong thing earlier) it looks like "Points History" overflows on the drawer -- should we make the font size smaller for everything?

File: navigation/AppNavigator.js [hunk 1]
```diff
@@ -1,99 +1,63 @@
...
+  return (
+    <NavigationContainer>
+      <AppStack.Navigator
+        initialRouteName="AuthLoading"
+        screenOptions={{
+          headerShown: false,
+          cardStyle: { backgroundColor: Colors.lightest },
+        }}>
+        <AppStack.Screen name="AuthLoading" component={AuthLoadingScreen} />
+        <AppStack.Screen name="App" component={DrawerNavigator} />
```
Comment (wangannie) [inline navigation/AppNavigator.js:58]: When this boots up, you can see it transition/move left from a blank screen to the map. Can you disable animations here to get rid of that?

File: navigation/stack_navigators/StoresStack.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    headerMode: 'none',
+  },
+});
+
+const StoresStack = createStackNavigator();
+
+export default function StoresStackNavigator() {
+  return (
+    <StoresStack.Navigator
+      screenOptions={{
```
Comment (wangannie) [inline navigation/stack_navigators/StoresStack.js:26]: Gestures don't work in Android (I think they're disabled by default) so we need to explicitly enable it here (mostly for the downward swiping interaction for the rewards page).

**Comment by anniero98** [comment]
+1 on `ProductsScreen`; this might be a todo for later, but also getting a warning
```
Non-serializable values were found in the navigation state, which can break usage such as persisting and restoring state. This might happen if you passed non-serializable values such as function, class instances etc. in params. If you need to use components with callbacks in your options, you can use 'navigation.setOptions' instead. See https://reactnavigation.org/docs/troubleshooting#i-get-the-warning-non-serializable-values-were-found-in-the-navigation-state for more details.
```
in StoreListScreen because it uses a callback function to change the "current store" in `MapScreen`

File: screens/map/ProductsScreen.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default class ProductsScreen extends React.Component {
   }
 
   renderProductList = () => {
-    const { products, store } = this.props.navigation.state.params;
+    const { products, store } = this.props.route.params;
```
[the remark this answers is not in the document]
Comment (thumn) [reply screens/map/ProductsScreen.js:20]: Yes!! Oops can't believe I missed this

[the remark this answers is not in the document]
Comment (thumn) [reply screens/map/ProductsScreen.js:20]: Oh oops it turns out I may have reverted this line because of a merge conflict -- will update it

File: navigation/AppNavigator.js [hunk 4]
```diff
@@ -1,99 +1,63 @@
...
+      <Drawer.Screen
+        name="Stores"
+        component={StoresStackNavigator}
+        options={{ title: 'Stores' }}
+      />
+      <Drawer.Screen
+        name="Rewards"
+        options={{ title: 'Points History', drawerLockMode: 'locked-closed' }}>
+        {props => <RewardsScreen {...props} tab={1} />}
+      </Drawer.Screen>
```
[the remark this answers is not in the document]
Comment (thumn) [reply navigation/AppNavigator.js:36]: Ok, I'll comment out the "Points History" drawer item!

[the remark this answers is not in the document]
Comment (wangannie) [reply navigation/AppNavigator.js:36]: Update: Kenneth updated in #74

File: navigation/stack_navigators/StoresStack.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        cardOverlayEnabled: true,
+        drawerLabel: 'Stores',
```
[the remark this answers is not in the document]
Comment (thumn) [reply navigation/stack_navigators/StoresStack.js:28]: `drawerLabel` in the stack itself and `title` in the drawer navigator both refer to what displays in the drawer for that item, but I think if `drawerLabel` isn't provided, it defaults to `title`. `name` is what we use to refer to the component when navigating between screens

File: navigation/stack_navigators/StoresStack.js (hunk 2 above)
Comment (thumn) [reply navigation/stack_navigators/StoresStack.js:26]: Hopefully this is fixed by my response to your previous comment about gesture direction!

File: navigation/AppNavigator.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (thumn) [reply navigation/AppNavigator.js:58]: weird...I'm unable to reproduce this right now but can you try again with my new fixes?

File: navigation/AppNavigator.js [hunk 6]
```diff
@@ -21,19 +20,21 @@ function DrawerNavigator() {
         labelStyle: {
           fontFamily: 'poppins-medium',
           fontSize: 20,
+          color: Colors.black,
```
Comment (wangannie) [inline navigation/AppNavigator.js]: Should be activetext
